### ISSUE TITLE
Upgrade from 21.6.3 to 21.9.0 fails

### ISSUE THREAD

**Issue by Sebi94nbg** [getsentry/sentry#29043]
### Version

21.6.3

### Steps to Reproduce

1. Have a running Sentry 21.6.3
2. `git checkout tags/21.9.0`
3. Ensure, the `.env` file contains the respective values as in Git
4. Ensure, all manually changed files are containing the latest changes as in Git (e.g. new variables or options in config or script files)
5. Run `./install.sh --no-user-prompt` to perform the actual Sentry upgrade

Additional Information:
```
$ git status
HEAD detached at 21.9.0
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   .env

no changes added to commit (use "git add" and/or "git commit -a")
```

```
$ git diff .env
diff --git a/.env b/.env
index aad7129..26c437a 100644
--- a/.env
+++ b/.env
@@ -1,12 +1,14 @@
+SENTRY_SECRET_KEY=VGhpcy...BJcyBVcwo=
 COMPOSE_PROJECT_NAME=sentry_onpremise
-SENTRY_EVENT_RETENTION_DAYS=90
+SENTRY_EVENT_RETENTION_DAYS=60
+
 # You can either use a port number or an IP:PORT combo for SENTRY_BIND
 # See https://docs.docker.com/compose/compose-file/#ports for more
-SENTRY_BIND=9000
+SENTRY_BIND=80
 # Set SENTRY_MAIL_HOST to a valid FQDN (host/domain name) to be able to send emails!
-# SENTRY_MAIL_HOST=example.com
+SENTRY_MAIL_HOST='sentry@example.com'
 SENTRY_IMAGE=getsentry/sentry:21.9.0
 SNUBA_IMAGE=getsentry/snuba:21.9.0
 RELAY_IMAGE=getsentry/relay:21.9.0
 SYMBOLICATOR_IMAGE=getsentry/symbolicator:0.3.4
-WAL2JSON_VERSION=latest
\ No newline at end of file
+WAL2JSON_VERSION=latest
```

### Expected Result

All works as expected and Sentry has been successfully upgraded to the respective version 21.9.0.

### Actual Result

The installation script is failing and Sentry is broken:
```
[...]
▶ Setting up / migrating database ...
Starting sentry_onpremise_zookeeper_1 ... 
Starting sentry_onpremise_symbolicator_1 ... 
Starting sentry_onpremise_redis_1        ... 
Starting sentry_onpremise_clickhouse_1   ... 
Starting sentry_onpremise_memcached_1    ... 
Starting sentry_onpremise_postgres_1     ... 
Starting sentry_onpremise_smtp_1         ... 
Starting sentry_onpremise_symbolicator_1 ... done
Starting sentry_onpremise_postgres_1     ... done
Starting sentry_onpremise_zookeeper_1    ... done
Starting sentry_onpremise_redis_1        ... done
Starting sentry_onpremise_memcached_1    ... done
Starting sentry_onpremise_clickhouse_1   ... done
Starting sentry_onpremise_smtp_1         ... done
Starting sentry_onpremise_kafka_1        ... 
Starting sentry_onpremise_kafka_1        ... done
Starting sentry_onpremise_snuba-outcomes-consumer_1 ... 
Starting sentry_onpremise_snuba-api_1               ... 
Starting sentry_onpremise_snuba-replacer_1          ... 
Starting sentry_onpremise_snuba-subscription-consumer-transactions_1 ... 
Starting sentry_onpremise_snuba-transactions-consumer_1              ... 
Starting sentry_onpremise_snuba-sessions-consumer_1                  ... 
Starting sentry_onpremise_snuba-subscription-consumer-events_1       ... 
Starting sentry_onpremise_snuba-consumer_1                           ... 
Starting sentry_onpremise_snuba-outcomes-consumer_1                  ... done
Starting sentry_onpremise_snuba-subscription-consumer-transactions_1 ... done
Starting sentry_onpremise_snuba-replacer_1                           ... done
Starting sentry_onpremise_snuba-sessions-consumer_1                  ... done
Starting sentry_onpremise_snuba-api_1                                ... done
Starting sentry_onpremise_snuba-transactions-consumer_1              ... done
Starting sentry_onpremise_snuba-consumer_1                           ... done
Starting sentry_onpremise_snuba-subscription-consumer-events_1       ... done
Creating sentry_onpremise_web_run                                    ... 
Creating sentry_onpremise_web_run                                    ... done
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/__init__.py", line 191, in main
    func(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/decorators.py", line 28, in inner
    configure()
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/__init__.py", line 127, in configure
    configure(ctx, py, yaml, skip_service_validation)
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/settings.py", line 155, in configure
    skip_service_validation=skip_service_validation,
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/initializer.py", line 353, in initialize_app
    django.setup()
  File "/usr/local/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.6/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/usr/local/lib/python3.6/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/sentry_interface_types/models.py", line 2, in <module>
    from sentry.plugins.bases.tag import TagPlugin
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/bases/__init__.py", line 1, in <module>
    from .issue import IssueTrackingPlugin  # NOQA
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/bases/issue.py", line 6, in <module>
    from sentry.plugins.base.v1 import Plugin
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/base/__init__.py", line 1, in <module>
    from .bindings import BindingManager
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/base/bindings.py", line 1, in <module>
    from sentry.plugins import providers
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/dummy/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/dummy/repository.py", line 1, in <module>
    from sentry.plugins.providers.repository import RepositoryProvider
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/repository.py", line 7, in <module>
    from sentry.api.serializers import serialize
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/__init__.py", line 2, in <module>
    from .models import *  # NOQA
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/models/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/models/alert_rule.py", line 6, in <module>
    from sentry.incidents.logic import translate_aggregate_field
  File "/usr/local/lib/python3.6/site-packages/sentry/incidents/logic.py", line 12, in <module>
    from sentry.incidents import tasks
  File "/usr/local/lib/python3.6/site-packages/sentry/incidents/tasks.py", line 26, in <module>
    from sentry.utils.email import MessageBuilder
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 246, in <module>
    options.get("mail.list-namespace"), default_list_type_handlers
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 205, in __init__
    assert is_valid_dot_atom(namespace)
AssertionError
1
An error occurred, caught SIGERR on line 4
Cleaning up...
```

Looks like as this is caused by the following script: `install/set-up-and-migrate-database.sh`

When I execute the command from this script manually, it's throwing the same error:

```
# docker-compose run --rm web upgrade
Starting sentry_onpremise_redis_1        ... done
Starting sentry_onpremise_memcached_1    ... done
Starting sentry_onpremise_postgres_1     ... done
Starting sentry_onpremise_clickhouse_1   ... done
Starting sentry_onpremise_symbolicator_1 ... done
Starting sentry_onpremise_zookeeper_1    ... done
Starting sentry_onpremise_smtp_1         ... done
Starting sentry_onpremise_kafka_1        ... done
Starting sentry_onpremise_snuba-api_1                                ... done
Starting sentry_onpremise_snuba-subscription-consumer-events_1       ... done
Starting sentry_onpremise_snuba-outcomes-consumer_1                  ... done
Starting sentry_onpremise_snuba-sessions-consumer_1                  ... done
Starting sentry_onpremise_snuba-subscription-consumer-transactions_1 ... done
Starting sentry_onpremise_snuba-replacer_1                           ... done
Starting sentry_onpremise_snuba-transactions-consumer_1              ... done
Starting sentry_onpremise_snuba-consumer_1                           ... done
Creating sentry_onpremise_web_run                                    ... done
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Traceback (most recent call last):
  File "/usr/local/bin/sentry", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/__init__.py", line 191, in main
    func(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/decorators.py", line 28, in inner
    configure()
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/__init__.py", line 127, in configure
    configure(ctx, py, yaml, skip_service_validation)
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/settings.py", line 155, in configure
    skip_service_validation=skip_service_validation,
  File "/usr/local/lib/python3.6/site-packages/sentry/runner/initializer.py", line 353, in initialize_app
    django.setup()
  File "/usr/local/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.6/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/usr/local/lib/python3.6/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/sentry_interface_types/models.py", line 2, in <module>
    from sentry.plugins.bases.tag import TagPlugin
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/bases/__init__.py", line 1, in <module>
    from .issue import IssueTrackingPlugin  # NOQA
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/bases/issue.py", line 6, in <module>
    from sentry.plugins.base.v1 import Plugin
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/base/__init__.py", line 1, in <module>
    from .bindings import BindingManager
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/base/bindings.py", line 1, in <module>
    from sentry.plugins import providers
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/dummy/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/dummy/repository.py", line 1, in <module>
    from sentry.plugins.providers.repository import RepositoryProvider
  File "/usr/local/lib/python3.6/site-packages/sentry/plugins/providers/repository.py", line 7, in <module>
    from sentry.api.serializers import serialize
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/__init__.py", line 2, in <module>
    from .models import *  # NOQA
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/models/__init__.py", line 3, in <module>
    import_submodules(globals(), __name__, __path__)
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/imports.py", line 42, in import_submodules
    module = __import__(module_name, globals(), locals(), ["__name__"])
  File "/usr/local/lib/python3.6/site-packages/sentry/api/serializers/models/alert_rule.py", line 6, in <module>
    from sentry.incidents.logic import translate_aggregate_field
  File "/usr/local/lib/python3.6/site-packages/sentry/incidents/logic.py", line 12, in <module>
    from sentry.incidents import tasks
  File "/usr/local/lib/python3.6/site-packages/sentry/incidents/tasks.py", line 26, in <module>
    from sentry.utils.email import MessageBuilder
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 246, in <module>
    options.get("mail.list-namespace"), default_list_type_handlers
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 205, in __init__
    assert is_valid_dot_atom(namespace)
AssertionError
ERROR: 1
```

**Comment by BYK** [comment]
```
File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 246, in <module>
    options.get("mail.list-namespace"), default_list_type_handlers
  File "/usr/local/lib/python3.6/site-packages/sentry/utils/email.py", line 205, in __init__
    assert is_valid_dot_atom(namespace)
```

Looks like `mail.list-namespace` setting does not have a valid email address in it?

**Comment by getsentry-release** [comment]
Routing to @getsentry/open-source for [triage](https://develop.sentry.dev/processing-tickets/#3-triage). ⏲️

**Comment by Sebi94nbg** [comment]
```
# grep "mail.from" sentry/sentry.conf.py 
SENTRY_OPTIONS["mail.from"] = f"sentry@{SENTRY_OPTIONS['mail.list-namespace']}"
```

```
$ grep ^SENTRY_MAIL_HOST= .env
SENTRY_MAIL_HOST='sentry@example.com'
```

Before, `mail.from` looked like this in our setup:
```
# grep "mail.from" sentry/sentry.conf.py 
mail.from: '{{ sentry_mail_smarthost_from_address | default('sentry@example.com') }}'
```

That's why I'm kinda confused about these changes and the new "host" variable, which should be an "email address" variable. :thinking:

**Comment by BYK** [comment]
@Sebi94nbg ah, this change was introduced here: https://github.com/getsentry/onpremise/pull/1076/files#r696092836

The value should be set in your `config.yml` file or in the options database. Can you check your options file?

**Comment by Sebi94nbg** [comment]
Ah, ok. Good, I got it working after running `docker-compose up -d` twice as the first attempt failed.

Necessary change to fix my issue:
```
-SENTRY_MAIL_HOST='{{ sentry_mail_smarthost_from_address | default('sentry@example.com') }}'
+SENTRY_MAIL_HOST='{{ inventory_hostname }}'
```

Sentry has been upgraded to 21.9.0 (4e50898) now. Thank you! :)

Is there somewhere a button, where I can test my SMTP settings, so that I don't have to wait for an automated e.g. weekly report?

I've found this page: https://<sentry.url>/manage/status/mail/

My user has the "Owner" role, but the page is reporting the following message:

> Your role does not have the necessary permissions to access this resource, please read more about organizational roles

Is this outdated or is there something broken in my installation?
